README.md

```diff
@@ -84,6 +86,70 @@ or have `InfoDialog` in their behavior text.
 Any option after the double dash (`--`) gets sent to `mocha`. Check out
 [mocha's usage](https://mochajs.org/#usage) for ideas.
 
+### Flow
+
+There is limited support for using [Flow](https://flowtype.org/)
+to check for problems in the source code.
+
+To check for Flow issues during development while you edit files, run:
+
+    npm run flow:dev
+
+If you are new to working with Flow, here are some tips:
+* Check out the [getting started](https://flow.org/en/docs/getting-started/) guide.
+* Read through the [web-ext guide](https://github.com/mozilla/web-ext/blob/master/CONTRIBUTING.md#check-for-flow-errors)
+  for hints on how to solve common Flow errors.
+
+To add flow coverage to a source file, put a `/* @flow */` comment at the top.
+The more source files you can opt into Flow, the better.
+
+Here is our Flow manifesto:
+
+* We use Flow to **declare the intention of our code** and help others
+  refactor it with confidence.
+  Flow also makes it easier to catch mistakes before spending hours in a debugger
+  trying to find out what happened.
+* Avoid magic [Flow declarations](https://flowtype.org/en/docs/config/libs/)
+  for any *internal* code. Just declare a
+  [type alias](https://flowtype.org/en/docs/types/aliases/) next to the code
+  where it's used and
+  [export/import](https://flow.org/en/docs/types/modules/) it like any other object.
+* Never import a real JS object just to reference its type. Make a type alias
+  and import that instead.
+* Never add more type annotations than you need. Flow is really good at
+  inferring types from standard JS code; it will tell you
+  when you need to add explicit annotations.
+* When a function like `getAllAddons` takes object arguments, call its
+  type object `GetAllAddonsParams`. Example:
+
+````js
+type GetAllAddonsParams = {|
+  categoryId: number,
+|};
+
+function getAllAddons({ categoryId }: GetAllAddonsParams = {}) {
+  ...
+}
+````
+
+* Use [Exact object types](https://flowtype.org/en/docs/types/objects/#toc-exact-object-types)
+  via the pipe syntax (`{| key: ... |}`) when possible. Sometimes the
+  spread operator triggers an error like
+  'Inexact type is incompatible with exact type' but that's a
+  [bug](https://github.com/facebook/flow/issues/2405).
+  You can use the `Exact<T>` workaround from
+  [`src/core/types/coreTypes`](https://github.com/mozilla/addons-frontend/blob/master/src/core/types/coreTypes.js)
```

Comment:
Why do we call the file `coreTypes` when it's already in the `types/` dir? I think I at least meant to ask but forgot or it got buried. Seems redundant though.